### PR TITLE
[feat][broker][PIP-195] Add metrics for bucket delayed message tracker

### PR DESCRIPTION
**Pull request by coderzc**
PIP: #16763

### Motivation

Add metrics for bucket delayed message tracker.

### Modifications

The new metrics and change metrics:

| **Name**                                                | **Type**  | **Description**                                              |
| :------------------------------------------------------ | --------- | ------------------------------------------------------------ |
| pulsar_delayed_message_index_bucket_total               | Gauge     | The number of delayed message index buckets (immutable buckets + LastMutableBucket ) |
| pulsar_delayed_message_index_loaded                     | Gauge     | The total number of delayed message indexes for in the memory. |
| pulsar_delayed_message_index_bucket_snapshot_size_bytes | Gauge     | The total size of delayed message index bucket snapshot (in bytes). |
| pulsar_delayed_message_index_bucket_op_count            | Counter   | The total number of operate delayed message index bucket snapshot. The `state` label can be `succeed`,`failed`,`all` (the `all` means is the total number of all states) and the `type` label can be `create`,`load`,`delete`,`merge`. |
| pulsar_delayed_message_index_bucket_op_latency_ms       | Histogram | The latency of delayed message index bucket snapshot operation with a given quantile (threshold). The label`type` label can be `create`,`load`,`delete`,`merge`<br/>The label `quantile` can be:<ul><li>quantile="50" is operation latency between (0ms, 50ms]</li><li>quantile="100" is operation latency between (50ms, 100ms]</li><li>quantile="500" is operation between (100ms, 500ms]</li><li>quantile="1000" is operation latency between (500ms, 1s]</li><li>quantile="5000" is operation latency between (1s, 5s]</li><li>quantile="30000" is operation latency between (5s, 30s]</li><li>quantile="60000" is operation latency between (30s, 60s]</li><li>quantile="overflow" is operation latency > 1m</li></ul> |

* Remove `containsMessage` from DelayedDeliveryTracker interface
* Remove `closeLedger` after `getLedgerEntry`

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/StatsBuckets.java [hunk 1]
```diff
@@ -73,6 +73,19 @@ public void refresh() {
...
+        long count = 0;
+        sum = sumCounter.sum();
+
+        for (int i = 0; i < buckets.length; i++) {
+            long value = buckets[i].sum();
+            count += value;
+            values[i] = value;
+        }
+
+        this.count = count;
```
Comment (dao-jun) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/StatsBuckets.java:86]: There seems to be a problem. For Histogram, `count` is the times of add/observe method invoked

**Review by mattisonchao — APPROVED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BookkeeperBucketSnapshotStorage.java [hunk 2]
```diff
@@ -212,8 +208,8 @@ private CompletableFuture<Void> addEntry(LedgerHandle ledgerHandle, byte[] data)
         });
     }
 
-    CompletableFuture<Enumeration<LedgerEntry>> getLedgerEntryThenCloseLedger(LedgerHandle ledger,
-                                                                              long firstEntryId, long lastEntryId) {
+    CompletableFuture<Enumeration<LedgerEntry>> getLedgerEntry(LedgerHandle ledger,
```
Comment (coderzc) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/bucket/BookkeeperBucketSnapshotStorage.java:211]: Ledger is already closed after `createBucketSnapshot`, so we don't need to close it again.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/DelayedDeliveryTracker.java [hunk 3]
```diff
@@ -67,12 +67,6 @@ public interface DelayedDeliveryTracker extends AutoCloseable {
      */
     boolean shouldPauseAllDeliveries();
 
-    /**
-     * Tells whether this DelayedDeliveryTracker contains this message index,
-     * if the tracker is not supported it or disabled this feature also will return false.
-     */
-    boolean containsMessage(long ledgerId, long entryId);
```
[the remark this answers is not in the document]
Comment (coderzc) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/DelayedDeliveryTracker.java:74]: Oh, sorry this addresses a previous comment.
See: https://github.com/apache/pulsar/pull/17677#pullrequestreview-1178729635

[the remark this answers is not in the document]
Comment (michaeljmarshall) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/delayed/DelayedDeliveryTracker.java:74]: Thanks for the context!

**Pull request merged**